### ISSUE TITLE
regarding integration of CNN model with GP model

### ISSUE THREAD

**Issue by irum** [cornellius-gp/gpytorch#1215]
I am loading the checkpoints from pretrianed (transfer learning ) resnet50 model. I get features from CNN model and use these features as input to the GP model like (pseudo code)

              cnn_model= resnet50
              load checkpoint
              cnn_model.fc = nn.Sequential(
                             nn.Linear(2048, 4096),
                             nn.ReLU(inplace=True),
                             nn.Dropout(p=0.5),
                             nn.Linear(4096,4096)).to(device)

in training loop

       def  train():
                     cnn_features = cnn_model(images)
                     cnn_features= cnn_features.double()
                     output = gp_model(cnn_features)
                     loss= -mll(output, target)

for GPclassification class : 

     class GPClassificationModel(ApproximateGP):
             def __init__(self, inducing_points,num_features):
             variational_distribution = CholeskyVariationalDistribution(inducing_points.size(0))
             variational_strategy = VariationalStrategy(self, inducing_points, variational_distribution, learn_inducing_locations=True)
             variational_strategy = gpytorch.variational.MultitaskVariationalStrategy(
            variational_strategy, num_tasks=num_features,
              )
            super(GPClassificationModel, self).__init__(variational_strategy)
            self.mean_module = gpytorch.means.ConstantMean()
            self.covar_module = gpytorch.kernels.ScaleKernel(
            gpytorch.kernels.SpectralMixtureKernel(num_mixtures=10, ard_num_dims=4096)
        
            )

           def forward(self,x):
               mean_x = self.mean_module(x)
               covar_x = self.covar_module(x)
              latent_pred = gpytorch.distributions.MultivariateNormal(mean_x, covar_x)
          return latent_pred
 
          gp_model = GPClassificationModel(inducing_points=inducing_pts,num_features=num_features).double()
          if mixing_weights==True:
                  likelihood = gpytorch.liklihhos.Softmaxlikelihood(num_classes=num_classes, mixing_weights=True, 
                  num_features=num_features).double()
          else:
                  likelihood = gpytorch.likelihoods.SoftmaxLikelihood(num_classes=num_classes,mixing_weights=False).double()
  
      
        
          gp_model = gp_model.to(device)
          likelihood = likelihood.to(device)
           mll = VariationalELBO(likelihood, gp_model, len(train_loader.dataset))

The problem is that I am getting constant error . It means gp model is not learning.

If I use the method defined in the [example](https://docs.gpytorch.ai/en/v1.1.1/examples/06_PyTorch_NN_Integration_DKL/Deep_Kernel_Learning_DenseNet_CIFAR_Tutorial.html)

Then how to write the code for  feature_extractor line for my cnn_model

**Comment by irum** [comment]
I have fixed my issue.

**Comment by QueeneTam** [comment]
Hi. Would you mind share that how did you finally fixed this issue?

**Comment by QueeneTam** [comment]
> I have fixed my issue.

Hi. Recently I met the same problem. Would you mind share that how did you finally fixed this issue? Thank you!
